src/libsyntax_expand/expand.rs

```diff
@@ -1031,6 +1032,7 @@ impl<'a, 'b> InvocationCollector<'a, 'b> {
         let features = self.cx.ecfg.features.unwrap();
         for attr in attrs.iter() {
             feature_gate::check_attribute(attr, self.cx.parse_sess, features);
+            validate_attr::check_meta(self.cx.parse_sess, attr);
```

Comment:
@petrochenkov  N.B. this diff. There shouldn't be any observable change from the POV of `expand.rs`. If there were, you'd see things removed in `src/test/ui/proc-macro/attribute.stderr`. This is indeed necessary to resolve a cyclic dependency, but I think this is also more appropriate for `feature_gate` to not do non-feature-gate related things.

Reply:
Ah, I see, didn't pay attention to this.